modules/imgcodecs/src/grfmt_avif.cpp

```diff
@@ -170,6 +158,20 @@ bool AvifDecoder::checkSignature(const String &signature) const {
     }                                                     \
   }
 
+bool AvifDecoder::checkSignature(const String &signature) const {
+  avifDecoder *decoder = avifDecoderCreate();
+  if (!decoder) return false;
+  OPENCV_AVIF_CHECK_STATUS(
+      avifDecoderSetIOMemory(
+          decoder, reinterpret_cast<const uint8_t *>(signature.c_str()),
+          signature.size()),
+      decoder);
+  decoder->io->sizeHint = 1e9;
+  const avifResult status = avifDecoderParse(decoder);
+  avifDecoderDestroy(decoder);
```

Comment:
Won't `decoder` instance leak if we exit earlier due to failed `avifDecoderSetIOMemory` call?

Reply:
You are totally right, I provided a fix in https://github.com/opencv/opencv/pull/24441. Thx !